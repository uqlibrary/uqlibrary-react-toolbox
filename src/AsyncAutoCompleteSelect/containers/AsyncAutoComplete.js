import React from 'react';
import PropTypes from 'prop-types';
import AsyncAutoCompleteSelect from '../components/AsyncAutoCompleteSelect.component';

import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

const AsyncAutoCompleteSelectWrapper = props => {
    const filteredProps = propFilter(props, AsyncAutoCompleteSelect.propTypes);
    return (
        <div style={{position: 'relative', width: '100%'}}>
            <AsyncAutoCompleteSelect {...filteredProps} />
            {props.helpText && (
                <div style={{position: 'absolute', right: 0, top: 0}}>
                    <HelpIcon title={props.helpTitle} text={props.helpText} buttonLabel="Got it!" />
                </div>
            )}
        </div>
    );
};

AsyncAutoCompleteSelectWrapper.propTypes = {
    ...AsyncAutoCompleteSelect.propTypes,
    helpTitle: PropTypes.string,
    helpText: PropTypes.any
};

export default AsyncAutoCompleteSelectWrapper;
