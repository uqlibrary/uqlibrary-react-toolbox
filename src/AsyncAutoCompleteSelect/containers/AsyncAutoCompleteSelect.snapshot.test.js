jest.dontMock('./AsyncAutoComplete');

import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import {default  as AsyncAutoComplete} from './AsyncAutoComplete';

function setup() {
    const props = {
        helpTitle: 'title',
        helpText: 'text',
        disabled: false,
        label: 'test'
    };

    return shallow(<AsyncAutoComplete {...props} />);
}


describe('Add Journal article form snapshot tests', () => {
    it('renders default add journal article form', () => {
        const app = setup();
        expect(toJson(app)).toMatchSnapshot();
    });
});
