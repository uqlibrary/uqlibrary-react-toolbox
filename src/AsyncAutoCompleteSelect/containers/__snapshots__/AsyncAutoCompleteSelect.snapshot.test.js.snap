// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Journal article form snapshot tests renders default add journal article form 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <AsyncAutoCompleteSelect
    alwaysRenderSuggestions={false}
    dataSourceLabel="label"
    disabled={false}
    errorText=""
    label="test"
    minLength={2}
    value={null}
  />
  <div
    style={
      Object {
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Connect(HelpIcon)
      buttonLabel="Got it!"
      text="text"
      title="title"
    />
  </div>
</div>
`;
